Jenkinsfile

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent { label 'executor-v2' }
+
+  options {
+    timestamps()
+    buildDiscarder(logRotator(numToKeepStr: '30'))
+  }
+
+  stages {
+    stage('Run tests') {
+      steps {
+        sh './test.sh'
+
+        junit 'features/reports/**/*.xml, spec/reports/*.xml'
+      }
+    }
+
+    // Only publish to RubyGems if the HEAD is
+    // tagged with the same version as in version.rb
+    stage('Publish to RubyGems') {
+      agent { label 'releaser-v2' }
+
+      when {
+        expression { currentBuild.resultIsBetterOrEqualTo('SUCCESS') }
+        expression {
+          def exitCode = sh returnStatus: true, script: ''' set +x
```

Comment:
@jvanderhoof was having trouble with this logic when used in the CLI. Can you check with him and see how he resolved his issue, please?

Reply:
yes, I tagged him as well so he could look at this. I did make some updates this morning to better mimic what is in the `conjur-cli` Jenkinsfile at current.